src/main/scala/cromwell/engine/callactor/CallActor.scala

```diff
@@ -74,15 +74,16 @@ object CallActor {
 
   val CallCounter = Monitor.minMaxCounter("calls-running")
 
-  def props(key: BackendCallKey, locallyQualifiedInputs: CallInputs, backend: Backend, workflowDescriptor: WorkflowDescriptor): Props =
-    Props(new BackendCallActor(key, locallyQualifiedInputs, backend, workflowDescriptor))
-  def props(key: FinalCallKey) = Props(new FinalCallActor(key))
+  def props(backendCallDescriptor: BackendCallDescriptor): Props = Props(new BackendCallActor(backendCallDescriptor))
+  def props(finalCallDescriptor: FinalCallDescriptor) = Props(new FinalCallActor(finalCallDescriptor))
 }
 
 /** Actor to manage the execution of a single call. */
-trait CallActor extends LoggingFSM[CallActorState, CallActorData] with CromwellActor {
-  def key: CallKey
-  def workflowDescriptor: WorkflowDescriptor
+// FIXME It feels like I shouldn't need to restate the type bounds of CallDescriptor's CallKey type variable.
+trait CallActor[D <: CallDescriptor[_ <: CallKey]] extends LoggingFSM[CallActorState, CallActorData] with CromwellActor {
```

Comment:
someone immersed in higher kindedness could probably straighten this out @geoffjentry